game/board.rb

```diff
@@ -0,0 +1,222 @@
+require "Matrix"
+
+class Board
+
+  WATER = "-"
+  SHIP = "S"
+  HIT = "X"
+  MISS = "O"
+
+
+  attr_reader :length, :height, :matrix
+
+  ################################################
+  # initialize
+  #
+  # initializes a m x n board as a 2D array
+  #
+  # m - the length
+  # n - the height
+  ###############################################
+  def initialize m, n
```

Comment:
For readability, consider avoiding single character argument names like this. Better to be more verbose:
```rb
def initialize length, height
  @length = length
  @height = height
  @matrix = Array.new(@length) { Array.new(@height, "-")}
```